### PR TITLE
Add find lesson command

### PR DESCRIPTION
**Pull request by ethantyh692**

### CODE REVIEW

File: src/main/java/tutorease/address/logic/commands/FindLessonCommand.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * Keyword matching is case-insensitive.
+ */
+public class FindLessonCommand extends Command {
+
+    public static final String COMMAND_WORD = "find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all lessons whose students' names "
+            + "match the specified names (case-insensitive) and displays them in the lesson panel. \n "
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " alice bob charlie";
```
Comment (NorbertLoh) [inline src/main/java/tutorease/address/logic/commands/FindLessonCommand.java:21]: Should the message usage be lesson find? If not we will have duplicates.

**Review by NorbertLoh — CHANGES_REQUESTED** [review]

File: src/main/java/tutorease/address/logic/commands/FindLessonCommand.java (hunk 1 above)
Comment (ethantyh692) [reply src/main/java/tutorease/address/logic/commands/FindLessonCommand.java]: Fixed

**Review by NorbertLoh — APPROVED** [review]
LGTM!

File: src/main/java/tutorease/address/logic/commands/FindLessonCommand.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                        model.getFilteredLessonList().size()));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
```
Comment (Ma-Xizi) [inline src/main/java/tutorease/address/logic/commands/FindLessonCommand.java:48]: I like how it aligns with the equals method of other commands and maintains coherency.

File: src/main/java/tutorease/address/model/lesson/LessonContainsNamesPredicate.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        LessonContainsNamesPredicate otherLessonContainsNamesPredicate =
+                (LessonContainsNamesPredicate) other;
+        return keywords.equals(otherLessonContainsNamesPredicate.keywords);
+    }
+
+    @Override
+    public String toString() {
+        return new ToStringBuilder(this).add("keywords", keywords).toString();
```
Comment (Ma-Xizi) [inline src/main/java/tutorease/address/model/lesson/LessonContainsNamesPredicate.java:43]: Maybe the toString() method could be more specific, to differentiate it from NameContainsKeyword class used in find contact command.

For instance, 

    @Override
    public String toString() {
        return new ToStringBuilder(this).add("keywords in lessons", keywords).toString();

Comment (ethantyh692) [reply src/main/java/tutorease/address/model/lesson/LessonContainsNamesPredicate.java]: Good idea. Fixed.

**Comment by Ma-Xizi** [comment]
LGTM!

**Pull request merged**